### ISSUE TITLE
Let params override baseParams

### ISSUE THREAD

**Issue by mplatvoet** [kittinunf/Fuel#16]
I think it makes more sense to let params override baseParams instead of the current implementation which defaults to baseParams if present. For me baseParams define the default but I would like to make an exeception in specific calls. Thus instead of doing `params + baseParams` I think it should be `baseParams + params`. 

What doe you think?

**Comment by kittinunf** [comment]
Thank you so much for your suggestion!!!! I am glad that you like Fuel.

I am not sure by having `params` overrides `baseParams`. I can explain why I opt to go for this option. 

My server wants to have `api_key` bundled in all API call in order to grant the access right. 

Consider, we have `baseParams=("api_key" to "abcdefghi")`

Then, API to list all dogs call will need to be 
`http://example.com/api/dogs?api_key=abcdefghi`

This means that if we make `params` overrides `baseParams`.  List all dogs with french bulldog type is supposedly be `http://example.com/api/dogs?api_key=abcdefghi&type=french_bulldog` which in this case `params=("type" to "french_bulldog")`. 

Then, if we let params overrides it, one might find it hard to use. So, my ultimate option is to add them together (both `params` and `baseParams`) to achieve use case that I am looking for.

If you have a better alternative, I am very happy to discuss.

**Comment by mplatvoet** [comment]
What I mean is when we have duplicate parameters, which should take precedence? E.g.
```kt
fun main(args: Array<String>) {
    val a = mapOf ("key" to 1)
    val b = mapOf ("key" to 2)
    val ab = a + b
    val ba = b + a
    println("ab = ${ab["key"]}") // prints 2
    println("ba = ${ba["key"]}") // prints 1
}
``` 
I think it makes more sense to let params take precedence over baseParams. That way you can configure for instance a default api_key as you suggest and let this be overridden by any concrete call.

Another point, which is maybe even *more important*, http allows you to have multiple values for the same key, so this:
`http://somedomain.com/?key=1&key=2&key=4`
is perfectly valid.

**Comment by kittinunf** [comment]
Oh!!! Sorry @mplatvoet for my stupidity!! In that case, I am totally on your side. To be honest, I overlooked this and agree with you that `params` should take precedence. It should be very simple. I will fix in the next release for sure. 

Also,happy to merge if you are willing to send PR to fix that. :smile:

**Comment by mplatvoet** [comment]
Yeah next time I can suggest these things with a PR. Last time I tried to build Fuel I got as far as running the tests which all failed at the time. But I believe you have fixed that now so I can give it another try :)

**Comment by kittinunf** [comment]
Yup. I fixed it now. Can you try build on `master` branch.

**Comment by kittinunf** [comment]
Fixed with version 0.52. Thanks for your valuable feedback. :+1:
